src/aiida_quantumespresso/workflows/pdos.py

```diff
@@ -458,7 +458,13 @@ def inspect_nscf(self):
         self.ctx.nscf_emin = workchain.outputs.output_band.get_array('bands').min()
         self.ctx.nscf_emax = workchain.outputs.output_band.get_array('bands').max()
         self.ctx.nscf_parent_folder = workchain.outputs.remote_folder
-        self.ctx.nscf_fermi = workchain.outputs.output_parameters.dict.fermi_energy
+        if 'fermi_energy' in workchain.outputs.output_parameters.dict:
+            self.ctx.nscf_fermi = workchain.outputs.output_parameters.dict.fermi_energy
+        else:
+            self.ctx.nscf_fermi_energy_up = workchain.outputs.output_parameters.dict.fermi_energy_up
```

Comment:
Thanks @AndresOrtegaGuerrero! Quick question here: do we need to keep `nscf_fermi_energy_up` and `nscf_fermi_energy_down` in the context if we determine the Fermi level from the highest value in line 466?

Reply:
Yes i will update this !